### PR TITLE
fix bug #1013

### PR DESCRIPTION
**Pull request by yvchan**
- in addition, honour SERVICE_NAME if it's set in the environment and no other value has been found.

### CODE REVIEW

**Comment by ddebrunner** [comment]
Is the name of the env variable really `SERVICE_NAME` or is it the existing variable `STREAMING_ANALYTICS_SERVICE_NAME` ?

File: java/src/com/ibm/streamsx/topology/internal/streaminganalytics/VcapServices.java [hunk 1]
```diff
@@ -22,82 +22,104 @@
...
     /**
      * Get the top-level VCAP services object.
      * 
-     * Object can be:
-     * JsonObject - assumed to contain VCAP_SERVICES
-     * String -  assumed to contain serialized VCAP_SERVICES JSON
-     * File - assumed to be a file containing serialized VCAP_SERVICES JSON
+     * Object can be one of the following:
+     * <ul>
+     * <li>JsonObject - assumed to contain VCAP_SERVICES, and SERVICE_NAME</li>
```
Comment (ddebrunner) [inline java/src/com/ibm/streamsx/topology/internal/streaminganalytics/VcapServices.java]: I'm confused by what this means, getVcapServices() just returns the vcap services without any setting of the service name to use.

File: java/src/com/ibm/streamsx/topology/internal/streaminganalytics/VcapServices.java [hunk 2]
```diff
@@ -22,82 +22,104 @@
...
         JsonArray streamsServices = array(services, "streaming-analytics");
         if (streamsServices == null || streamsServices.size() == 0)
             throw new IllegalStateException("No streaming-analytics services defined in VCAP_SERVICES");
-        
+
         String serviceName = jstring(deploy, SERVICE_NAME);
 
+        // if we don't find our serviceName check the environment variable
+        if (serviceName == null) {
+            serviceName = System.getenv("SERVICE_NAME");
```
Comment (ddebrunner) [inline java/src/com/ibm/streamsx/topology/internal/streaminganalytics/VcapServices.java]: Needs to match the existing variable name.

File: java/src/com/ibm/streamsx/topology/internal/streaminganalytics/VcapServices.java (hunk 1 above)
Comment (yvchan) [inline java/src/com/ibm/streamsx/topology/internal/streaminganalytics/VcapServices.java]: i'll take out the "and SERVICE_NAME" bit..   this is returning the whole set of credentials & names in the object.

File: java/src/com/ibm/streamsx/topology/internal/streaminganalytics/VcapServices.java (hunk 2 above)
Comment (yvchan) [inline java/src/com/ibm/streamsx/topology/internal/streaminganalytics/VcapServices.java]: i will change `SERVICE_NAME `to `STREAMING_ANALYTICS_SERVICE_NAME`

**Pull request merged**